src/components/app.tsx

```diff
@@ -8,45 +8,58 @@ import { DonationEntry } from '../types/global.d'
 
 import 'bootstrap/dist/css/bootstrap.css';
 
-let updErrorData: React.Dispatch<React.SetStateAction<string | null>>;
-let updIsLoaded: React.Dispatch<React.SetStateAction<boolean>>;
-let updData: React.Dispatch<React.SetStateAction<DonationEntry[]>>;
+interface TableState {
+  loaded: boolean,
+  error: string | null,
+  data: DonationEntry[],
+}
 
-const fetchData: Function = () => {
-  fetch("https://inlight-panda-rescue-api.herokuapp.com/donations?apiKey=cr2eJJDmDK94NgbaPL8Z")
+const fetchData: () => Promise<TableState> = () => {
+  return fetch("https://inlight-panda-rescue-api.herokuapp.com/donations?apiKey=cr2eJJDmDK94NgbaPL8Z")
     .then(res => res.json())
     .then((result) => {
-      updIsLoaded(true);
-      updData(result);
+      const stateObj: TableState = { loaded: true, data: result, error: null }
+      return stateObj
     }, (error) => {
-      updIsLoaded(true);
-      updErrorData(error);
+      const stateObj: TableState = { loaded: true, data: [], error: error }
+      return stateObj
     })
 }
 
 const App: React.FC = () => {
-  const [errorData, setErrorData] = useState<string | null>(null);
-  const [isLoaded, setIsLoaded] = useState<boolean>(false);
-  const [data, setData] = useState<DonationEntry[]>([]);
-
-  updErrorData = setErrorData;
-  updIsLoaded = setIsLoaded;
-  updData = setData;
+  const [state, setState] = useReducer(
```

Comment:
I'm confused as to why we've replaced with reducer to solve for pure functions. Can we please put back `useState`? (at least for now)